SSINS/sky_subtract.py

```diff
@@ -173,6 +193,32 @@ def diff(self):
         super().set_lsts_from_time_array()
         self.data_array = np.ma.masked_array(self.data_array)
 
+    def diff_freq(self):
+
+        """
+        Differences the visibilities in freq. Does so independently for each time.
+        The flags are propagated by taking the boolean OR of the entries that correspond
+        to the visibilities that are differenced from one another. Other metadata
+        attributes are also adjusted so that the resulting SS object passes
+        UVData.check()
+        """
+
+        if self.blt_order != 'baseline':
+            warnings.warn("Reordering data array to baseline order to perform differencing.")
+            self.reorder_blts(order='baseline')
+
+        diff_dat = np.diff(self.data_array, axis=2) #axis 2: nfreqs (see uvdata object)
+        self.data_array = diff_dat
+
+        print(self.data_array.shape)
+        """The frequency-differenced visibilities. Complex array of shape (Nblts, Nspws, Nfreqs, Npols)."""
+
+        self.Nfreqs -= 1
+        self.data_array = np.ma.masked_array(self.data_array)
+        self.freq_array = (self.freq_array[:,1:] + self.freq_array[:,:-1]) / 2.0
+        self.nsample_array = self.nsample_array[:,:,1:,:]
```

Comment:
Should average the nsample_array (same operation as is done on the freq_array)

Reply:
Fixed in 9ed2f618642c06563768b9944f11fd265413a0b1